docs/proposals/container-runtime-interface/capabilities.md

```diff
@@ -0,0 +1,221 @@
+        // unless the pod spec specifically drops the capability or the pod spec uses a requested-set.
+        // You may not list a capability in both DefaultAddCapabilities and RequiredDropCapabilities.
+        // Note: this field is deprecated in favor of the DefaultCapabilities field.
+        DefaultAddCapabilities []api.Capability `json:"defaultAddCapabilities,omitempty"`
+
+        // RequiredDropCapabilities are the capabilities that will be dropped from the container. These
+        // are required to be dropped and cannot be added or present in the requested-set.
+        RequiredDropCapabilities []api.Capability `json:"requiredDropCapabilities,omitempty"`
+
+        // AllowedCapabilities is a list of capabilities that can be requested to add to the container.
+        // Capabilities in this field may be added at the pod author's discretion via the add
+        // or requested-set in the pod spec.
+        // You must not list a capability in both AllowedCapabilities and RequiredDropCapabilities.
+        AllowedCapabilities []api.Capability `json:"allowedCapabilities,omitempty"`
+
+        // DefaultCapabilities is a set of default capabilities that will be used for pods
+        // that do not have a requested-set specified in their pod spec.
+        // The pod spec's add/drop capabilities are applied to this set.
+        // In addition to the capabilities listed here, the pod author can add additional capabilities
+        // specified in the AllowedCapabilities field.
+        DefaultCapabilities []api.Capability `json:"defaultCapabilities,omitempty"`
+...
+}
+```
+
+The `Capabilities` struct in `pkg/api/types.go`, and `pkg/api/v1/types.go` gets a new `RequestedSet` field:
+
+```
+// Capabilities represent POSIX capabilities that can be added or removed to a running container to/from a default set.
+type Capabilities struct {
+        // Capabilities added to the default capabilities
+        Add []Capability `json:"add,omitempty"`
+        // Capabilities removed from the default capabilities
+        Drop []Capability `json:"drop,omitempty"`
+        // Requested set of capabilities
+        RequestedSet []Capability `json:"requestedSet,omitempty"`
+}
+```
+
+The `Capability` struct in `pkg/kubelet/api/v1alpha1/runtime/api.proto` replaces the existing `add_capabilities` and `drop_capabilities` fields with a new `requested_capabilities` field:
+
+```
+message Capability {
+    // List of capabilities to retain.
+    repeated string requested_capabilities = 1;
+}
+```
+
+## Use Cases
+
+### User specifies `requestedSet` only in the pod spec:
+
+The simplest use case is when a user only sets the `requestedSet` field in a new pod spec:
+
+```
+  spec:
+    containers:
+      securityContext:
+        capabilities:
+          requestedSet:
+          - MKNOD, NET_ADMIN, SETPCAP, AUDIT_WRITE, ...
+```
+
+The resolved requested set equals `requestedSet`.
```

Comment:
Ack, I documented this in the godoc of `PodSecurityPolicySpec` struct but not in this paragraph.

Indeed `DefaultAddCapabilities` is silently ignored, I will add a sentence to clarify this.

Regarding `RequiredDropCapabilities` I somewhat sloppy specified that "these cannot be added or present in the requested-set". Semantically I consider this a validation error, even a violation of the pod security policy, since the user requests caps that have to be dropped. I'm adding a sentence to clarify this.